### PR TITLE
Farcaster auth

### LINKED ISSUE #10194: Integrate farcaster into ProfileV3
follow up on profileV3

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [Koda - Generative Art Marketplace](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Feature

## Needs Design check

- @exezbcz please review

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #10194

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Comment by JustLuuuu** [comment]
I signed the message in the FC app. I think it was successful, but nothing happened on Koda. 

https://github.com/kodadot/nft-gallery/assets/104380776/90819245-aa82-4747-9050-d90bb797db35

**Comment by vikiival** [comment]
Sir please have you tested? 

1. Login modal will never close (prob bug of that modal?)

I am stuck here

Screenshot 2024-05-03 at 12 19 00

I need to supply both image and banner on my onw

Screenshot 2024-05-03 at 12 19 21

**Review by vikiival — CHANGES_REQUESTED** [review]
as per test

**Comment by daiagi** [comment]
>Sir please have you tested?


yes. i did
 now will test some more :smile:

**Comment by daiagi** [comment]
@JustLuuuu 

you are supposed to scan the QR and the log in via phone

**Comment by daiagi** [comment]
oh ok. it doesn't work in preview env.

worked in local-host

:bug:

**Comment by daiagi** [comment]
@vikiival 

small yet critical bug

fixed

image

**Comment by daiagi** [comment]
one thing

even importing farcaster profile **Still Have to** upload banner picture

do we want to change that @vikiival  @JustLuuuu ?

**Comment by vikiival** [comment]
cc @exezbcz 

I would make banner optional for now :)

**Comment by vikiival** [comment]
> you are supposed to scan the QR and the log in via phone

Screenshot 2024-05-03 at 15 21 02

**Comment by prury** [comment]
this qr code that warpcast provides is unreadable for me haha, had to send the whole link via telegram and access it from mobile, then it worked, not our problem i know.

regarding the whole functionality, agree with banner being optional
and will open another issue for for fields being optional or not.

**Comment by daiagi** [comment]
> optional banner

requires updates to worker, ill take care of it

>Login modal will never close (prob bug of that modal?)

sorry @vikiival  could you rephrase plz? are you talking about the farcaster QR tab?

**Comment by vikiival** [comment]
> sorry @vikiival  could you rephrase please? are you talking about the farcaster QR tab?

Yes

How does it work now:
1. Scan the QR
2. Allow login 
3. Nothing happens - eg i need to close the tab 

The reason why @JustLuuuu was confused

How it should work

1. Scan the QR
2. Allow login
3. Tab/modal should be close so i can see my profile modal

**Comment by daiagi** [comment]
@vikiival  @prury 

video below show working

https://github.com/kodadot/nft-gallery/assets/22791238/b2c019eb-1620-4ff8-aa29-7cf75cd4c420

**Comment by vikiival** [comment]
https://github.com/kodadot/nft-gallery/assets/22471030/07184889-0d35-414d-abfc-912aef097c77


Screenshot 2024-05-03 at 19 57 06

Screenshot 2024-05-03 at 19 57 06

**Review by vikiival — CHANGES_REQUESTED** [review]
> Nothing to preview

**Comment by daiagi** [comment]
Yes

Is being handled separately here


https://github.com/kodadot/private-workers/issues/137

**Comment by vikiival** [comment]
Soo can we test?

**Comment by daiagi** [comment]
@vikiival 

works without banner

https://profile-beta.kodadot.workers.dev/profiles/5DWh36v5AbEQ9FXWPRK9prPuA8bZMoZoJJ9PvA3DnL6i85zF

image

File: components/profile/create/Modal.vue [hunk 1]
```diff
@@ -121,4 +125,51 @@ const handleFormSubmition = async (profileData: ProfileFormData) => {
...
+      return
+    }
+    if (userData.data.nonce !== channel.data.nonce) {
+      console.error('nonce mismatch')
+      return
+    }
+    farcasterUserData.value = userData.data
+  } else {
+    console.error('URL not found in channel data')
+  }
```
Comment (Jarsen136) [inline components/profile/create/Modal.vue:173]: How about showing a notification toast for these errors?

Comment (exezbcz) [reply components/profile/create/Modal.vue:173]: > How about showing a notification toast for these errors?

nice catch! makes sense, already created one for reminding that the action is on new tab

- https://github.com/kodadot/nft-gallery/issues/10234

Comment (daiagi) [reply components/profile/create/Modal.vue:173]: Sure

Let's do this in #10234

**Review by Jarsen136 — APPROVED** [review]

**Comment by vikiival** [comment]
Screenshot 2024-05-06 at 12 17 46

**Comment by vikiival** [comment]
Description is using different component that collection description (a.k.a MD does not work)

Screenshot 2024-05-06 at 12 20 11


Edit does not work.

**Comment by vikiival** [comment]
But auth works, others can be done in the follow-up

**Review by vikiival — APPROVED** [review]
